L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloEGammaEmulator.cc

```diff
@@ -0,0 +1,753 @@
+                                      1.1639, 1.1639,
+                                      1.1639, 1.1639,
+                                      1.1742, 1.1742,
+                                      1.1639, 1.1639,
+                                      1.1742, 1.1639,
+                                      1.1639, 1.1742,
+                                      1.1742, 1.1536000000000002,
+                                      1.11,   1.11,
+                                      1.11,   1.11,
+                                      1.11,   1.11,
+                                      1.11,   1.11,
+                                      1.11,   1.11,
+                                      1.11,   1.11,
+                                      1.11,   1.11,
+                                      1.11,   1.11,
+                                      1.1,    1.09,
+                                      1.09,   1.09,
+                                      1.09,   1.09,
+                                      1.09,   1.09,
+                                      1.09,   1.09,
+                                      1.09,   1.09,
+                                      1.09,   1.09,
+                                      1.09,   1.09,
+                                      1.09,   1.09,
+                                      1.07,   1.07,
+                                      1.07,   1.07,
+                                      1.07,   1.07,
+                                      1.07,   1.08,
+                                      1.07,   1.07,
+                                      1.08,   1.08,
+                                      1.07,   1.08,
+                                      1.08,   1.08,
+                                      1.08,   1.06,
+                                      1.06,   1.06,
+                                      1.06,   1.05,
+                                      1.05,   1.06,
+                                      1.06,   1.06,
+                                      1.06,   1.06,
+                                      1.06,   1.06,
+                                      1.06,   1.06,
+                                      1.06,   1.06,
+                                      1.04,   1.04,
+                                      1.04,   1.04,
+                                      1.05,   1.04,
+                                      1.05,   1.05,
+                                      1.05,   1.05,
+                                      1.05,   1.05,
+                                      1.05,   1.05,
+                                      1.05,   1.05,
+                                      1.05,
+                                  });
+    desc.add<edm::ParameterSetDescription>("calib", psd0);
+  }
+  descriptions.add("l1tPhase2L1CaloEGammaEmulator", desc);
+  // or use the following to generate the label from the module's C++ type
+  //descriptions.addWithDefaultLabel(desc);
```

Comment:
As such, you end up with two configuration files named l1tPhase2L1CaloEGammaEmulator_cfi.py, and both accessible in python as L1Trigger.L1CaloTrigger.l1tPhase2L1CaloEGammaEmulator_cfi. This generates confusion (especially if at some time any of them differs from the other) and must be avoided.

If these configurations are not expected to change, you can remove the explicit file L1Trigger/L1CaloTrigger/python/l1tPhase2L1CaloEGammaEmulator_cfi.py

Since I imagine that is not the case, then:
```suggestion
  descriptions.addWithDefaultLabel(desc);
```

After that, you can use this configuration description only to validate the one in L1Trigger/L1CaloTrigger/python/l1tPhase2L1CaloEGammaEmulator_cfi.py

Or, you can replace the l1tPhase2L1CaloEGammaEmulator configuration in it with
```PYTHON
from L1Trigger.L1CaloTrigger.phase2L1CaloEGammaEmulator_cfi import phase2L1CaloEGammaEmulator
l1tPhase2L1CaloEGammaEmulator = phase2L1CaloEGammaEmulator.clone()
```
where possible modified parameters can be written as arguments of the clone()

Reply:
@perrotta I have @skkwan sitting in my office, and we want to clarify a point of preference here.

To my understanding right now the `_cfi` does/did two things. First it configured (redundantly before the change) the EGamma emulator, but second, we have this era modification hanging around:

```python
from Configuration.ProcessModifiers.premix_stage2_cff import premix_stage2
premix_stage2.toModify(l1tPhase2L1CaloEGammaEmulator,
    ecalTPEB = "DMEcalEBTriggerPrimitiveDigis",
    hcalTP = "DMHcalTriggerPrimitiveDigis",
)
```

After looking at this for a minute, I think there are two possible ways this end up:

```python
import FWCore.ParameterSet.Config as cms

from L1Trigger.L1CaloTrigger.phase2L1CaloEGammaEmulator_cfi import phase2L1CaloEGammaEmulator
l1tPhase2L1CaloEGammaEmulator = phase2L1CaloEGammaEmulator.clone(    
    ecalTPEB = "DMEcalEBTriggerPrimitiveDigis",
    hcalTP = "DMHcalTriggerPrimitiveDigis",
)
```

And then we expect that whoever uses the phase2_premix era uses this module instead of the auto-generated `phase2L1CaloEGammaEmulator`. This seems a bit prone to failure. Alternatively, I guess one could do this:

```python
import FWCore.ParameterSet.Config as cms

from L1Trigger.L1CaloTrigger.phase2L1CaloEGammaEmulator_cfi import phase2L1CaloEGammaEmulator

from Configuration.ProcessModifiers.premix_stage2_cff import premix_stage2
premix_stage2.toModify(phase2L1CaloEGammaEmulator,
    ecalTPEB = "DMEcalEBTriggerPrimitiveDigis",
    hcalTP = "DMHcalTriggerPrimitiveDigis",
)
```

This should maintain the era modifications for `phase2L1CaloEGammaEmulator`, but this no longer seems like a `_cfi` per se. Should this be relegated to some customization, or other python file? What's the preference for cleaning this up? Looking at the `phase2_premix` era, most files seem to have `.toModify()` to change it in some `_cfi`.